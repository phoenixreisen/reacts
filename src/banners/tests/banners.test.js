/**
 * @jest-environment jsdom
 */
import {render, fireEvent, waitFor, screen} from '@testing-library/react';
import renderer from 'react-test-renderer';
import ShareBanner from '../share.r';
import LinkBanner from '../link.r';
import React from 'react';

describe('Link Banner should', () => {

    test('render correctly', () => {
        const component = renderer.create(
            <LinkBanner
                text="Hallo"
                key="linkbanner"
                urltext="Phoenix Reisen"
                url="https://www.phoenixreisen.com"
            />
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('throw error without params', () => {
        const out = console.error;
        console.error = () => {};

        let error = null;
        let error2 = null;

        // Without params at all
        try { render(<LinkBanner />); }
        catch(e) { error = e; }

        // Just without "text" param
        try { renderer.create(<LinkBanner url="url" urltext="text" />); }
        catch(e) { error2 = e; }

        expect(error).not.toBeNull();
        expect(error2).toBeNull();
        console.error = out;
    });
});

describe('Share Banner should', () => {

    test('render correctly without params', () => {
        const component = renderer.create(<ShareBanner />);
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('use given params correctly', () => {
        const component = renderer.create(
            <ShareBanner
                appname="Phoenix Reisen"
                mailsubject="Mail Subject"
                headline="Sie finden uns auch auf"
                urltext="Tolle Reisen mit dem Schiff"
                hashtags="Phoenix, Reisen, Kreuzfahrten"
            />
        );
        const snap = component.toJSON();
        expect(snap).toMatchSnapshot();
    });

    test('be with background by default', () => {
        const component = render(<ShareBanner />);
        expect(document.querySelector('.share-banner--bg-less')).toBeNull();
    });
    test('be backgroundless when `noBackground` is `true`', () => {
        const component = render(<ShareBanner noBackground={true} />);
        expect(document.querySelector('.share-banner--bg-less')).not.toBeNull();
    });

    test('toggle clipboard icon on click', () => {
        const component = render(<ShareBanner />);

        expect(document.querySelector('.share-clipboard')).not.toBeNull();
        expect(document.querySelector('.fa-clipboard-check')).toBeNull();
        expect(document.querySelector('.share-clipboard--clipped')).toBeNull();

        fireEvent.click(document.querySelector('.share-clipboard'));
        expect(document.querySelector('.fa-clipboard-check')).not.toBeNull();
        expect(document.querySelector('.share-clipboard--clipped')).not.toBeNull();
    });
});