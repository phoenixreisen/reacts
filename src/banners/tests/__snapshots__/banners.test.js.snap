// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Banner should render correctly 1`] = `
<article
  className="link-banner mv5"
>
  <div
    className="link-banner__textbox"
  >
    <p>
      Hallo
    </p>
    <p
      className="pb0"
    >
      <a
        href="https://www.phoenixreisen.com"
      >
        <i
          className="fas fa-angle-right"
        />
         
        Phoenix Reisen
      </a>
    </p>
  </div>
</article>
`;

exports[`Share Banner should render correctly without params 1`] = `
<article
  className="share-banner "
>
  <div
    className="wrapper wrapper--large"
  >
    <div
      className="tc"
    >
      <h3
        className="tc pt0"
      >
        Gerne weitersagen...
      </h3>
    </div>
    <div
      className="tc share-icons"
    >
      <a
        className="share-email"
        href="mailto:?subject=tolle Empfehlung&body=: http://localhost/"
        title=" per Email empfehlen"
      >
        <i
          className="fas fa-envelope"
        />
      </a>
      <a
        className="share-whatsapp"
        href="https://api.whatsapp.com/send?text=: http://localhost/"
        title=" per WhatsApp empfehlen"
      >
        <i
          className="fab fa-whatsapp"
        />
      </a>
      <a
        className="share-twitter"
        href="https://twitter.com/intent/tweet?text=&url=http://localhost/&hashtags=undefined"
        title=" twittern"
      >
        <i
          className="fab fa-twitter"
        />
      </a>
      <a
        className="share-facebook"
        href="https://www.facebook.com/sharer/sharer.php?u=http://localhost/"
        rel="noopener noreferrer"
        target="_blank"
        title=" per Facebook empfehlen"
      >
        <i
          className="fab fa-facebook-f"
        />
      </a>
      <a
        className="share-clipboard share-clipboard"
        data-clipboard-text=": http://localhost/"
        href="#"
        onClick={[Function]}
        title="in die Zwischenablage kopieren"
      >
        <i
          className="fas fa-clipboard"
        />
      </a>
    </div>
  </div>
</article>
`;

exports[`Share Banner should use given params correctly 1`] = `
<article
  className="share-banner "
>
  <div
    className="wrapper wrapper--large"
  >
    <div
      className="tc"
    >
      <h3
        className="tc pt0"
      >
        Sie finden uns auch auf
      </h3>
    </div>
    <div
      className="tc share-icons"
    >
      <a
        className="share-email"
        href="mailto:?subject=Mail Subject&body=Tolle Reisen mit dem Schiff: http://localhost/"
        title="Phoenix Reisen per Email empfehlen"
      >
        <i
          className="fas fa-envelope"
        />
      </a>
      <a
        className="share-whatsapp"
        href="https://api.whatsapp.com/send?text=Tolle Reisen mit dem Schiff: http://localhost/"
        title="Phoenix Reisen per WhatsApp empfehlen"
      >
        <i
          className="fab fa-whatsapp"
        />
      </a>
      <a
        className="share-twitter"
        href="https://twitter.com/intent/tweet?text=Tolle Reisen mit dem Schiff&url=http://localhost/&hashtags=Phoenix, Reisen, Kreuzfahrten"
        title="Phoenix Reisen twittern"
      >
        <i
          className="fab fa-twitter"
        />
      </a>
      <a
        className="share-facebook"
        href="https://www.facebook.com/sharer/sharer.php?u=http://localhost/"
        rel="noopener noreferrer"
        target="_blank"
        title="Phoenix Reisen per Facebook empfehlen"
      >
        <i
          className="fab fa-facebook-f"
        />
      </a>
      <a
        className="share-clipboard share-clipboard"
        data-clipboard-text="Tolle Reisen mit dem Schiff: http://localhost/"
        href="#"
        onClick={[Function]}
        title="in die Zwischenablage kopieren"
      >
        <i
          className="fas fa-clipboard"
        />
      </a>
    </div>
  </div>
</article>
`;
